### PR TITLE
Record: Docstrings and doctestable examples for features and evaluated

### PR DESCRIPTION
**Pull request by mHash1m**
The Record class is still a WIP. I wanted to get a review on the functions before moving on to the next ones.
Signed-off-by: mHash1m <hashimchaudry23@gmail.com>

### CODE REVIEW

File: dffml/record.py [hunk 1]
```diff
@@ -182,6 +182,28 @@ def key(self) -> str:
     def evaluated(self, results: Dict[str, Any], overwrite=False):
         """
         Updates features with the result dict
+
+        Parameters
+        ----------
+        results : Dict
```
Comment (pdxjohnny) [inline dffml/record.py]: I think this needs to be lowercase

```suggestion
        results : dict
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Documentation on writing examples and running doctests
 - Doctestable Examples to high-level API.
 - Shouldi got an operation to run npm-audit on JavaScript code
+- Docstrings and doctestable examples for `record.py`(features and eveluated)
```
Comment (pdxjohnny) [inline CHANGELOG.md]:
```suggestion
- Docstrings and doctestable examples for `record.py` (features and eveluated)
```

File: dffml/record.py [hunk 3]
```diff
@@ -182,6 +182,28 @@ def key(self) -> str:
...
+        ...     "example",
+        ...     data=dict(
+        ...         features=dict(dead="beef"), extra=dict(extra="read all about it"),
+        ...     ),
+        ...     extra=dict(half=True),
+        ... )
+        >>>
+        >>> results = {"new": "feature"}
+        >>> example.evaluated({"feed": "face"})
+        >>> example.evaluated(results, overwrite=True)
```
[the remark this answers is not in the document]
Comment (mHash1m) [reply dffml/record.py]: Ah, makes sense.

File: dffml/record.py [hunk 4]
```diff
@@ -193,6 +215,30 @@ def evaluated(self, results: Dict[str, Any], overwrite=False):
...
+        Dict
+            features.
+
+        Examples
+        --------
+        >>> example = Record(
+        ...     "example",
+        ...     data=dict(
+        ...         features=dict(dead="beef"),
+        ...         extra=dict(extra="read all about it"),
```
[the remark this answers is not in the document]
Comment (mHash1m) [reply dffml/record.py]: I hope it's better now? Added the example of initialized data, adding to it, and overwriting it.

**Comment by mHash1m** [comment]
@pdxjohnny  Really sorry you had to go through all that trouble. :confused:

**Comment by pdxjohnny** [comment]
@mHash1m no trouble at all :) very nice! Thank you!

**Pull request merged**